### PR TITLE
Bundle webdriveragent pipeline

### PR DESCRIPTION
**Pull request by dpgraham**
Added a pipeline that bundles WebDriverAgents for every xcode version [see here](https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=3365).

Runs build.sh for every version of Xcode that Azure makes available (betas excluded), copies the `DerivedData` directory into the same directory as WebDriverAgent, creates a .tgz bundle, publishes the .tgz bundle to GitHub Releases.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,15 +38,16 @@
   "homepage": "https://github.com/appium/WebDriverAgent#readme",
   "devDependencies": {
     "appium-gulp-plugins": "^4.1.0",
+    "appium-support": "^2.29.0",
```
Comment (imurchie) [inline package.json]: This is already in `dependencies`. And is needed there. Same with `asyncbox`.

File: ci-jobs/scripts/build-webdriveragents.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  log.info(`Will skip xcode versions: '${excludedXcodeArr}'`);
+
+  for (let xcodePath of xcodePaths) {
+    if (xcodePath.includes('_beta')) {
+      continue;
+    }
+    // Build webdriveragent for this xcode version
+    log.info(`Running xcode-select for '${xcodePath}'`);
+    await exec('sudo', ['xcode-select', '-s', `/Applications/${xcodePath}/Contents/Developer`]);
+    const xcodeVersion = xcodePath.replace('.app', '').split('_', 2)[1];
```
Comment (mykola-mokhnach) [inline ci-jobs/scripts/build-webdriveragents.js]: path.parse(xcodePath).name

File: package.json (hunk 1 above)
Comment (mykola-mokhnach) [inline package.json]: it this change deliberate?

File: ci-jobs/scripts/build-webdriveragent.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+async function buildWebDriverAgent (xcodeVersion) {
+  // Get Xcode version
+  xcodeVersion = xcodeVersion || await xcode.getVersion();
+  log.info(`Building bundle for Xcode version '${xcodeVersion}'`);
+
+  // Clean and build
+  await exec('npx', ['gulp', 'clean:carthage']);
+  log.info('Running ./Scripts/build.sh');
+  await exec('./Scripts/build.sh');
```
Comment (mykola-mokhnach) [inline ci-jobs/scripts/build-webdriveragent.js]: does it work like this? I thought it should be done like `exec('/bin/bash', ['Scripts/build.sh'], {cwd: ...})`

**Review by imurchie — COMMENTED** [review]
This seems like a good approach. Hard to know how it will work without seeing the end product working.

File: ci-jobs/scripts/build-webdriveragents.js (hunk 2 above)
Comment (dpgraham) [reply ci-jobs/scripts/build-webdriveragents.js]: That won't get the version... I want to parse `version` from `xcode_<version>.app`

e.g.) `xcode_10.2.app` should parse as `10.2`

File: package.json (hunk 1 above)
Comment (dpgraham) [reply package.json]: No. I ran `npm install appium-support --save-dev` and it moved the dependency over to `devDependencies`.... I wish it wouldn't do that.

Comment (dpgraham) [reply package.json]: Fixed it.

File: ci-jobs/scripts/build-webdriveragent.js (hunk 3 above)
Comment (dpgraham) [reply ci-jobs/scripts/build-webdriveragent.js]: It does work but I'll change it anyway to be safe.

**Comment by dpgraham** [comment]
Pushed some more fixes.... I've tested it and it works locally. From here https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=3390 it saves the tarballs as artifacts. I was able to download the one that matched my local Xcode, set `usePrebuiltWDA`, and set `derivedDataPath` and it successfully started a session.

File: ci-jobs/scripts/build-webdriveragents.js (hunk 2 above)
Comment (dpgraham) [reply ci-jobs/scripts/build-webdriveragents.js]: nvm I understand now... changed it to `path.parse(xcodePath).split('_', 2)[1]`

File: ci-jobs/scripts/build-webdriveragent.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+async function buildWebDriverAgent (xcodeVersion) {
+  // Get Xcode version
+  xcodeVersion = xcodeVersion || await xcode.getVersion();
+  log.info(`Building bundle for Xcode version '${xcodeVersion}'`);
+
+  // Clean and build
+  await exec('npx', ['gulp', 'clean:carthage']);
+  log.info('Running ./Scripts/build.sh');
+  let env = {TARGET: 'runner', SDK: 'sim'};
+  await exec('/bin/bash', ['./Scripts/build.sh'], {env});
```
Comment (mykola-mokhnach) [inline ci-jobs/scripts/build-webdriveragent.js]: so what is the reason to not provide cwd explicitly?

Comment (dpgraham) [reply ci-jobs/scripts/build-webdriveragent.js]: The directory that `./Scripts/build.sh` needs to be run from is the same as the current working directory. Shouldn't we only be setting `cwd` if it's different from the actual current working directory?

Comment (mykola-mokhnach) [reply ci-jobs/scripts/build-webdriveragent.js]: nope, if we 100% sure that the current folder is always the root folder of WDA project

Comment (dpgraham) [reply ci-jobs/scripts/build-webdriveragent.js]: Okay. I changed it to use the root directory for cwd and path resolves.

**Comment by mykola-mokhnach** [comment]
it looks like CI is not very happy though

**Comment by dpgraham** [comment]
@mykola-mokhnach yeah I'm sorting that out now.

**Pull request merged**